theano/tensor/opt.py

```diff
@@ -2204,7 +2204,7 @@ def local_cast_cast(node):
 
     when those contrain:
     dtype1 == dtype2
-    TODO: the base dtype is the same (int, uint, float, complex)
+    OR the base dtype is the same (int, uint, float, complex)
           and the first cast cause an upcast.
 
     """
```

Comment:
Add this or similar:

Also remove single cast that cast to the original dtype. like:
x{fvector}.cast('float32') -> x

this was already done and missing from the comment.